cnf-tests/testsuites/e2esuite/ptp/ptp.go

```diff
@@ -3,157 +3,55 @@ package ptp
+		}
+		if fullConfig.PtpModeDesired != testconfig.Discovery {
+			ptptest.RestartPtpDaemon()
 		}
 
-		daemonset, err := client.Client.DaemonSets(ptpLinuxDaemonNamespace).Get(context.Background(), ptpDaemonsetName, metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		expectedNumber := daemonset.Status.DesiredNumberScheduled
-		Eventually(func() int32 {
-			daemonset, err = client.Client.DaemonSets(ptpLinuxDaemonNamespace).Get(context.Background(), ptpDaemonsetName, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			return daemonset.Status.NumberReady
-		}, 2*time.Minute, 2*time.Second).Should(Equal(expectedNumber))
-
-		Eventually(func() int {
-			ptpPods, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
-			Expect(err).ToNot(HaveOccurred())
-			return len(ptpPods.Items)
-		}, 2*time.Minute, 2*time.Second).Should(Equal(int(expectedNumber)))
-	})
-
-	var _ = Describe("Test Offset", func() {
-		slaveLabel := ptpSlaveNodeLabel
-		BeforeEach(func() {
-			if !discovery.Enabled() {
-				Skip("Offset test is enabled only in discovery mode, assuming the grandmaster is external to the cluster")
-			}
-			_, slaveConfigs := getPTPConfigs(ptpLinuxDaemonNamespace)
-			if len(slaveConfigs) == 0 {
-				Skip("No nodes configured as ptp slaves found on the cluster")
-			}
-			slaveLabel = retrievePTPProfileLabels(slaveConfigs)
-			if slaveLabel == "" {
-				Skip("No nodes configured as ptp slaves found on the cluster: no node with PTP slave labels found")
-			}
-		})
-
-		var _ = Context("PTP configuration verifications", func() {
-
-			// 27324
-			It("PTP time diff between Grandmaster and Slave should be in range -100ms and 100ms", func() {
-				var timeDiff string
-				ptpPods, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
-				Expect(err).ToNot(HaveOccurred())
-				Expect(len(ptpPods.Items)).To(BeNumerically(">", 0))
-				slavePodDetected := false
-				for _, pod := range ptpPods.Items {
-					if podRole(pod, slaveLabel) {
-						Eventually(func() string {
-							buf, _ := pods.ExecCommand(client.Client, pod, []string{"curl", "127.0.0.1:9091/metrics"})
-							timeDiff = buf.String()
-							return timeDiff
-						}, 3*time.Minute, 2*time.Second).Should(ContainSubstring("openshift_ptp_offset_ns{from=\"master\""),
-							fmt.Sprint("Time metrics are not detected"))
-						Expect(compareOffsetTime(timeDiff)).ToNot(BeFalse(), "Offset is not in acceptable range")
-						slavePodDetected = true
-					}
-				}
-				Expect(slavePodDetected).ToNot(BeFalse(), "No slave pods detected")
-			})
-		})
 	})
-
 	var _ = Describe("PTP socket sharing between pods", func() {
-		slaveLabel := ptpSlaveNodeLabel
+		slaveLabel := utils.PtpClockUnderTestNodeLabel
 		BeforeEach(func() {
-			if discovery.Enabled() {
+			if fullConfig.PtpModeDesired == testconfig.Discovery {
```

Comment:
I do not understand this comment, what do you mean by "discovery being a property of the execution" ?

Reply:
I don't know what is setting that ptpmodedesired variable, but if a user runs the suite with the related `DISCOVERY_MODE` environment variable set, he would expect these tests to run in discovery mode as well.

Reply:
yes, this is the current behavior, the legacy DISCOVERY_MODE is still honored

Reply:
going  forward, the new way to specify the test mode is PTP_TEST_MODE. This can be OC, BC, Dualnicbc or discovery. So right now we can run one test at a time by setting the PTP_TEST_MODE to the desired mode. How are you planning to run the 3 tests modes in succession? will cnf-feature-deploy run multiple times while setting the env variable to something else (maybe on a different setup). Or should the ptp-operator conformance tests run all three node (or subset of) on the same run?

Reply:
If the test mode is required to be set, then I'd say the user will have to run the tests focusing on ptp and iterating over the mode. Please also note that the user is now internal, as we are dismissing support of cnf-tests to RH users. This gives us a bit more flexiblity - clunkiness tolerance.